eos-enable-zram

```diff
@@ -13,13 +13,79 @@ if [ "$1" -eq 0 ]; then
     exit 0
 fi
 
-ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+# List and disable any active swap areas
+active_swaps=$(swapon --show=NAME --raw --noheadings)
+echo "Active swap areas before $(basename $0): ${active_swaps}"
+swapoff -a
 
+# Enable zram
+ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
 if [ ! -e /sys/block/zram0 ]; then
     modprobe zram
 fi
 echo $(($ram_size_kb * 2))K > /sys/block/zram0/disksize
-# For now, disable the swap partition if in use
-swapoff -a
 mkswap /dev/zram0
 swapon /dev/zram0
+
+if [ -z "${active_swaps}" ]; then
+    exit 0
+fi
+
+# Find root partition, root partition number, root disk, and table type
+orig_root_part=$(systemctl show -p What sysroot.mount)
+orig_root_part=${orig_root_part#What=}
+root_part=$(readlink -f "${orig_root_part}")
+case ${root_part} in
+    /dev/sd??)
+        root_disk=${root_part%?}
+        ;;
+    /dev/*p[0-9])
+        root_disk=${root_part%p?}
+        ;;
+esac
+
+case ${orig_root_part} in
+    /dev/mapper/endless-image?)
+        root_disk=${orig_root_part%?}
+        ;;
+esac
+
+if [ -z "${root_disk}" ]; then
+    exit 0
+fi
+
+get_last_char() {
```

Comment:
This is Bash so you don't need this monstrosity:

```
$ x=foobar
$ echo ${x: -1}
r
```

Reply:
(but maybe this is now a bit long to be a shell script anyway…)

Reply:
-ECOPYANDPASTE from a `/bin/sh` script.